course/src/main/java/org/openedx/course/presentation/container/CollapsingLayout.kt

```diff
@@ -0,0 +1,655 @@
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.draw.blur
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.input.nestedscroll.NestedScrollConnection
+import androidx.compose.ui.input.nestedscroll.NestedScrollSource
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.input.pointer.PointerEventType
+import androidx.compose.ui.input.pointer.PointerInputChange
+import androidx.compose.ui.input.pointer.PointerInputScope
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.openedx.core.ui.displayCutoutForLandscape
+import org.openedx.core.ui.rememberWindowSize
+import org.openedx.core.ui.statusBarsInset
+import org.openedx.core.ui.theme.appColors
+import kotlin.math.roundToInt
+
+@Composable
+internal fun CollapsingLayout(
+    courseImage: Bitmap,
+    expandedTop: @Composable BoxScope.() -> Unit,
+    collapsedTop: @Composable BoxScope.() -> Unit,
+    navigation: @Composable BoxScope.() -> Unit,
+    bodyContent: @Composable BoxScope.() -> Unit,
+    onBackClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val localDensity = LocalDensity.current
+    var expandedTopHeight by remember {
+        mutableFloatStateOf(0f)
+    }
+    var collapsedTopHeight by remember {
+        mutableFloatStateOf(0f)
+    }
+    var navigationHeight by remember {
+        mutableFloatStateOf(0f)
+    }
+    val offset = remember { Animatable(0f) }
+    var backgroundImageHeight by remember {
+        mutableFloatStateOf(0f)
+    }
+    val windowSize = rememberWindowSize()
+    val coroutineScope = rememberCoroutineScope()
+    val configuration = LocalConfiguration.current
+    val imageHeight = 200
```

Comment:
200 is "magic" number, maybe it's better to put it in arguments, or at least in consts